x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractorFactory.java

```diff
@@ -173,7 +180,7 @@ private static void validateIndexAndExtractFields(Client client,
                                                       ActionListener<ExtractedFields> listener) {
         AtomicInteger docValueFieldsLimitHolder = new AtomicInteger();
 
-        // Step 3. Extract fields (if possible) and notify listener
+        // Step 4. Extract fields (if possible) and notify listener
```

Comment:
Since the target field is a required field, we should probably verify that the index has all the fields required BEFORE we gather the cardinality limits.

Reply:
Done.
Moved this code into `ExtractedFieldsDetector` class.